specs/eventing/overview.md

```diff
@@ -1,97 +1,152 @@
-_Callable_ and _Addressable_ interface contracts, and writes the resolved
-addresses to the _Channel_ in the `channel` reference. _Subscriptions_ do not
-need to specify both a transformation and a storage destination, but at least
-one must be provided.
-
-All event delivery linkage from a **Subscription** is 1:1 – only a single
-`channel`, `subscriber`, and `reply` may be provided.
-
-For more details, see [Kind: Subscription](spec.md#kind-subscription).
-
-## Channel
-
-**Channel** provides an event delivery mechanism which can fan out received
-events to multiple destinations via _Subscriptions_. A _Channel_ has a single
-inbound _Addressable_ interface which may accept events delivered directly or
-forwarded from multiple _Subscriptions_. Different _Channels_ may implement
-different degrees of persistence. Event delivery order is dependent on the
-backing implementation of the _Channel_.
-
-Event selection on a _Channel_ is 1:N – a single _Channel_ may fan out to
-multiple _Subscriptions_.
-
-See [Kind: Channel](spec.md#kind-channel).
-
----
-
-_Navigation_:
-
-- [Motivation and goals](motivation.md)
-- **Resource type overview**
-- [Interface contracts](interfaces.md)
-- [Object model specification](spec.md)
+The Knative Eventing API provides primitives for two common event-processing
+patterns:
+
+- Topology-based event routing ([`messaging.knative.dev`](#messaging))
+
+  Events are routed based on _connections between objects_ (in particular,
+  events flow along a [channel](#channel) to all
+  [subscriptions](#subscription)). This model can be thought of as "event
+  plumbing" in that events are managed like flows of water through pipes. In
+  order to enable event processing, subscriptions provide a mechanism to route
+  "reply" events further through the object topology.
+
+- Content-based event routing ([`eventing.knative.dev`](#eventing))
+
+  Events are selected for routing based on the event _attributes_ rather than
+  primarily by object connections (a [broker](#broker) provides a stream of
+  events which can be selected by a [trigger](#trigger)). This model is more
+  akin to picking parts off a conveyor belt, where each event is considered
+  separately for processing. Content-based routing handles reply events by
+  re-enqueuing the events in the originating Broker.
+
+Knative Eventing does not directly specify mechanisms for other event-processing
+models, including multi-stage workflows, correlated request-reply, and
+sequential (windowed) event processing; these models could be built using the
+primitives provided by Knative, or Knative could deliver events to an external
+system that implements these models.
+
+In addition to the primitives needed to express the above patterns, Knative
+Eventing defines two [_interface contracts_](#interface-contracts) to allow
+connecting multiple types of Kubernetes objects as event senders and recipients
+to the core primitives.
+
+![Overview of objects](images/eventing-overview.svg)
```

Comment:
Should the picture show replies from the Trigger Subscriber going back to the Broker? It might help show this diff from the Messaging flow.